src/ConflictResolver/AutoIncrementConflictResolver.php

```diff
@@ -40,20 +42,32 @@ public function __construct(AdapterInterface $adapter)
      */
     public function resolveConflict(UriContext $uriContext)
     {
-        $this->inc = 0;
+        $this->index = 0;
+        $candidateUri = $uri = $uriContext->getUri();
 
-        $uri = $uriContext->getUri();
-        $candidateUri = $this->incrementUri($uri);
-
-        while ($route = $this->adapter->findRouteForUri($candidateUri, $uriContext)) {
+        while ($this->isUriConflicting($candidateUri, $uriContext)) {
             $candidateUri = $this->incrementUri($uri);
         }
 
         return $candidateUri;
     }
 
+    /**
+     * Tell if the given URI for the given context is conflicting with another
+     * route.
+     */
+    protected function isUriConflicting($uri, UriContext $uriContext)
+    {
+        if (null === $uriContext->getCollection()->getAutoRouteByUri($uri)
+            && null === $this->adapter->findRouteForUri($uri, $uriContext)) {
+            return false;
```

Comment:
please simplify this method to `return null !== $uriContext->getCollection()->getAutoRouteByUri($uri) || null !== ...findRouteForUri...;`